src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                              String dateFrom, String dateTo)
+            throws DateTimeParseException {
+        int wagePerHour;
+        int workingHours;
+        LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateEnd = LocalDate.parse(dateTo, formatter);
+        LocalDate workDayDate;
+        StringBuilder report = new StringBuilder("Report for period " + dateStart.format(formatter)
+                + " - " + dateEnd.format(formatter));
+        for (String name : names) {
+            int totalSalary = 0;
+            for (String datum : data) {
+                String[] record = datum.split(" ");
+                workDayDate = LocalDate.parse(record[0], formatter);
```

Comment:
```suggestion
                int workDayDate = LocalDate.parse(record[DATE_INDEX], formatter);
```
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/b2d4322c-5718-4533-94f7-3399d98cbb3d)

Reply:
Corrected